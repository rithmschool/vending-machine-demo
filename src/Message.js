import React from 'react';
import './Message.css';

const Message = ({ children }) => (
  <div id="messagey" className="Message">
    {children}
  </div>
);

export default Message;
