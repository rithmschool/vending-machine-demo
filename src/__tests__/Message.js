import React from 'react'; // for JSX
import { shallow } from 'enzyme'; // how to mount the component
import toJson from 'enzyme-to-json'; // for snapshotting the component
import Message from '../Message'; // import the component itself

describe('<Message />', () => {
  let wrapper;

  it('renders', () => {
    wrapper = shallow(<Message>hello world</Message>);
  });

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('has headings with proper text', () => {
    wrapper = shallow(
      <Message>
        <h1>I am nested</h1>
      </Message>
    );
    const h1 = wrapper.find('h1');
    expect(h1).toHaveLength(1);
    expect(h1.text()).toEqual('I am nested');
  });
});
