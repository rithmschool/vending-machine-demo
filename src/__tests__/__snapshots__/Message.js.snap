// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> matches snapshot 1`] = `
<div
  className="Message"
  id="messagey"
>
  hello world
</div>
`;
